aggie-coding-club/Rev-Registration
# Update saved courses when course cards are updated

## Description
Saved courses now continuously update as the user enters data. To do this, I created a `throttle` function to limit the number of API calls being made.

## Rationale
Couple of things I want to point out:
- if you close the page, your courses will be saved immediately in order to prevent data loss. However, this doesn't work as intended if you refresh, since the call to save courses will more than likely still be going on before the page is refreshed, and it will fetch slightly outdated courses. I'm not sure how we could fix this without compromising responsiveness, so any suggestions are welcome
- I didn't add any tests to make sure courses are saved as expected yet since I feel like we should all discuss when we should be saving in more detail first
- I made the timeout for throttling 15 seconds since that seemed reasonable, if we can't find a way around the issue with refreshing we might want to make the timer a bit shorter

## How to test
Change some course cards, and either close the page or change to a new term, your course cards should always be completely up-to-date. If you refresh, they revert to how they were when the throttle timer finished last (due to the issue I pointed out in [Rationale](#rationale).

## Related tasks
Closes #299
